bftengine/src/bftengine/ControllerWithSimpleHistory.hpp

```diff
@@ -66,6 +76,14 @@ class ControllerWithSimpleHistory : public ControllerBase {
   // Returns true in case a sequence number is within the range of the current window.
   bool insideActiveWindow(const SeqNum& n);
 
+  // Returns lower or upper bound, in case val is out of range, otherwise returns val.
+  template <typename T>
+  const T& normalizeToRange(const T& lower, const T& upper, const T& val) {
```

Comment:
The function is generic and technically can serve any types supporting operators `<` and `>`.
I think it might be good to make it free and move to `util` directory.

Reply:
Correct, for simplicity I left it there, but I'll move it to as a util

Reply:
I don't know if it will be used.
I thought about whether it's worth exporting it, my original choice was to leave it within the class.
Though it makes sense also to define it as util.

Reply:
I'll leave it in the class and define it static